text/0000-extended-standard-library.md

```diff
@@ -0,0 +1,370 @@
+
+### 3. Maintainer support
+
+Supporting maintainers of the software we use is a good thing to do. It can also improve the quality and reliability of the software by allowing for additional investment in the software and the processes around it, reducing burnout, and minimizing the pressure for software to change hands to potentially less trustworthy maintainers.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Bringing Rust closer to its goal, empowering everyone to build reliable and efficient software, means developers need:
+
+1. More confidence in the trustworthiness of the basic building blocks they rely on.
+2. An easier time finding the right building blocks.
+
+To achieve this, the Rust ecosystem needs a set of crates that includes the most commonly used functionality in Rust programs, referred to here as the ESL. It needs to be maintained by a trustworthy organization with a strong commitment to quality, security and support. As this set of crates matures, it should become possible to compose most Rust programs with a minimal number of dependencies outside of the ESL and the Rust standard library.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The ESL will consist of a set of crates providing commonly used functionality.
+
+## Policies
+
+The ESL will require a number of policies, the most essential of which will ensure timely security fixes, strike a thoughtful balance between API stability and evolution, provide long-term support for older compilers through a careful MSRV choice while avoiding stagnation, and maintain a high standard for quality. The ESL does not need to make the same policy choices as the Rust Standard Library.
+
+A simplified list of example policies to use as potential starting point can be found in [Appendix A](#appendix-a). Policies may need to cover additional topics beyond the examples provided.
+
+Policies should evolve as the ESL community learns about what works best.
+
+## External Dependencies
+
+In an ideal world the ESL would have no production (non-dev) dependencies outside of the ESL and the Rust standard library. However, this is likely going to be impractical because an existing set of widely used crates with dependencies on each other already exists. If one or more of these crates does not want to join the ESL, exceptions may need to be made allowing them to be external dependencies in order to avoid expending a huge amount of effort to create and maintain duplicates of their functionality for the ESL. The ESL community should work to minimize the number of external dependencies for the sake of offering the most consistent, secure, and reliable experience possible, but some will almost certainly be necessary.
```

Comment:
I worry this will be too restrictive. I'm splitting up `toml`. If it was in the ESL, I'd have to get those lower level crates to be blessed to a similar level which also unduely raises their visibility withinea blessed list.

I worry this can also be stagnating with other dependencies.

Reply:
we already have a policy distinguishing between intentional artifacts vs. internal use libs.
https://forge.rust-lang.org/policies/crate-ownership.html

Reply:
Another example: I'm working on a replacement for libtest.  It has its own cli parser and soon a json writer to meet very specific design requirements. Abandoning those requirements would be rough blessing each part would also.

In both examples, some of the packages are also being developed on an experimental basis. And the libtest project just left rust-lang org because of how strict the requirements are, even for experiments.

Reply:
It's unfortunately demonstrably true that, until we have multi-threaded rustc, it's faster for builds to split crates into multiple sub-crates as often as possible. We probably want to encourage large crates used by many people to break up their code as much as possible, rather than encouraging single large crates.

Reply:
> we already have a policy distinguishing between intentional artifacts vs. internal use libs.

The Project has that but this is separate.  That policy is also very strict in that something can only be Intentionally provided by the Project or its solely Internal to the project.  Some of the cases I'm referring to would be Intention for the ESL or non-blessed within the ESL but still Intentionally for end-users.